lib/Service/DraftsService.php

```diff
@@ -0,0 +1,240 @@
+			throw new ClientException('Cannot convert message to outbox message without at least one recipient');
+		}
+
+		$message = $this->mapper->updateWithRecipients($message, $toRecipients, $ccRecipients, $bccRecipients);
+
+		if (empty($attachments)) {
+			$message->setAttachments($this->attachmentService->updateLocalMessageAttachments($account->getUserId(), $message, []));
+			return $message;
+		}
+
+		$client = $this->clientFactory->getClient($account);
+		try {
+			$attachmentIds = $this->attachmentService->handleAttachments($account, $attachments, $client);
+		} finally {
+			$client->logout();
+		}
+		$message->setAttachments($this->attachmentService->updateLocalMessageAttachments($account->getUserId(), $message, $attachmentIds));
+		return $message;
+	}
+
+	public function handleDraft(Account $account, int $draftId): void {
+		$message = $this->mailManager->getMessage($account->getUserId(), $draftId);
+		$this->eventDispatcher->dispatchTyped(new DraftMessageCreatedEvent($account, $message));
+	}
+
+	/**
+	 * "Send" the message
+	 *
+	 * @param LocalMessage $message
+	 * @param Account $account
+	 * @return void
+	 */
+	public function sendMessage(LocalMessage $message, Account $account): void {
+		try {
+			$this->transmission->sendLocalMessage($account, $message, true);
+		} catch (ClientException|ServiceException $e) {
+			$this->logger->error('Could not move draft to IMAP', ['exception' => $e]);
+			// Mark as failed so the message is not moved repeatedly in background
+			$message->setFailed(true);
+			$this->mapper->update($message);
+			throw $e;
+		}
+		$this->attachmentService->deleteLocalMessageAttachments($account->getUserId(), $message->getId());
+		$this->mapper->deleteWithRecipients($message);
+	}
+
+	/**
+	 * @throws DoesNotExistException
+	 */
+	public function getMessage(int $id, string $userId): LocalMessage {
+		return $this->mapper->findById($id, $userId);
+	}
+
+	/**
+	 * @throws NotImplemented
+	 */
+	public function getMessages(string $userId): array {
+		throw new NotImplemented('Not implemented');
+	}
+
+	public function flush() {
+		$messages = $this->mapper->findDueDrafts($this->time->getTime());
+		if (empty($messages)) {
+			return;
+		}
+
+		$accountIds = array_unique(array_map(function ($message) {
+			return $message->getAccountId();
+		}, $messages));
+
+		$accounts = array_combine($accountIds, array_map(function ($accountId) {
+			try {
+				return $this->accountService->findById($accountId);
+			} catch (DoesNotExistException $e) {
+				// The message belongs to a deleted account
+				return null;
+			}
+		}, $accountIds));
+
+		foreach ($messages as $message) {
+			try {
+				$account = $accounts[$message->getAccountId()];
+				if ($account === null) {
+					// Ignore message of non-existent account
+					continue;
+				}
+				$this->sendMessage(
+					$message,
+					$account,
+				);
+				$this->logger->debug('Draft {id} moved to IMAP', [
+					'id' => $message->getId(),
+				]);
```

Comment:
this doesn't seem right. the message is moved as part of the sending process. but it shouldn't be sent

Reply:
It actually works because we pass a special flag to *send message* which prevents it from sending the message.